core/drivers/crypto/caam/include/caam_status.h

```diff
@@ -0,0 +1,26 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright 2018-2019 NXP
+ *
+ * Brief   CAAM driver internal status definition
+ */
+
+#ifndef __CAAM_STATUS_H__
+#define __CAAM_STATUS_H__
+
+/*
+ * Internal CAAM Driver status codes
+ */
+enum CAAM_Status {
```

Comment:
I'm wondering if this is needed or we could use `TEE_Result` instead.
If we're translating any of these values into `TEE_Result` values we could save a bit. There are other drivers that uses `TEE_Result` instead of a enum defined for just that particular driver.

Reply:
This is something we introduce following internal NXP security expert groups. This code is implemented since a long time and I'm reworking it for the nth time. If you really want me to remove that I can do. Please advise.

Reply:
This isn't very important. Right now you don't seem to translate this into different TEE_Result values very often. If you prefer keeping as it is, do so.